Tests/SourceGenerators/Realm.SourceGenerator.Tests/ComparisonTests.cs

```diff
@@ -41,19 +41,28 @@ public async Task ComparisonTest(string filename, params string[] classNames)
             await RunComparisonTest(filename, classNames);
         }
 
-        [TestCase("ClassWithBaseType")]
-        [TestCase("MultiplePrimaryKeys")]
-        [TestCase("NoPartialClass")]
-        [TestCase("RealmintegerErrors")]
-        [TestCase("RealmObjectAndEmbeddedObjectClass")]
-        [TestCase("UnsupportedIndexableTypes")]
-        [TestCase("UnsupportedPrimaryKeyTypes")]
-        [TestCase("UnsupportedRequiredTypes")]
-        [TestCase("NestedClassWithoutPartialParent")]
-        [TestCase("NullableErrorClass")]
-        [TestCase("IgnoreObjectNullabilityClass")]
-        [TestCase("UnsupportedBacklink", "UnsupportedBacklink", "BacklinkObj")]
-        public async Task ErrorComparisonTest(string filename, params string[] classNames)
+        public static object[][] ErrorTestCases = Directory.GetFiles(_errorClassesPath)
```

Comment:
I understand the sentiment, but you have to admit it's kinda ugly 😁

Reply:
Agree, but we clearly forget to add test cases there - the CollectionError tests were missing and I was wondering for quite some time why nothing is failing when I changed my code 😅

Reply:
I reworked the test discovery - let me know if now it's prettier 😜

Reply:
Regexes ftw it seems! It looks better I think ;)